paper/src/main/java/com/programmerdan/minecraft/simpleadminhacks/hacks/basic/MapCopyProtection.java

```diff
@@ -0,0 +1,126 @@
+package com.programmerdan.minecraft.simpleadminhacks.hacks.basic;
+
+import com.programmerdan.minecraft.simpleadminhacks.SimpleAdminHacks;
+import com.programmerdan.minecraft.simpleadminhacks.framework.BasicHack;
+import com.programmerdan.minecraft.simpleadminhacks.framework.BasicHackConfig;
+import net.kyori.adventure.text.Component;
+import net.kyori.adventure.text.format.TextColor;
+import org.bukkit.Bukkit;
+import org.bukkit.Material;
+import org.bukkit.NamespacedKey;
+import org.bukkit.command.Command;
+import org.bukkit.command.CommandExecutor;
+import org.bukkit.command.CommandSender;
+import org.bukkit.craftbukkit.v1_18_R2.inventory.CraftInventoryCrafting;
+import org.bukkit.entity.Player;
+import org.bukkit.event.EventHandler;
+import org.bukkit.event.inventory.InventoryClickEvent;
+import org.bukkit.event.inventory.InventoryType;
+import org.bukkit.inventory.CartographyInventory;
+import org.bukkit.inventory.ItemStack;
+import org.bukkit.inventory.meta.ItemMeta;
+import org.bukkit.persistence.PersistentDataType;
+import org.jetbrains.annotations.NotNull;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class MapCopyProtection extends BasicHack implements CommandExecutor {
+	final NamespacedKey copyKey = new NamespacedKey(plugin, "copy-protected");
+
+	public MapCopyProtection(SimpleAdminHacks plugin, BasicHackConfig config) {
+		super(plugin, config);
+	}
+
+	@Override
+	public void onEnable() {
+		super.onEnable();
+		plugin().registerCommand("copyprotect", this);
+	}
+
+	@Override
+	public void onDisable() {
+		super.onDisable();
+	}
+
+	@EventHandler
+	public void onMapCopyCartography(InventoryClickEvent event) {
+		if (!(event.getInventory() instanceof CartographyInventory) || event.getSlotType() != InventoryType.SlotType.RESULT)
+			return;
+
+		ItemStack item = event.getInventory().getItem(1); // paper, glass pane, blank map slot
+
+		if (item == null) return;
+
+		if (item.getType() == Material.GLASS_PANE || item.getType() == Material.PAPER) return; // Allow locking and scaling of copy protected maps
+
+		handleCopyProtection(event);
+	}
+
+	@EventHandler
+	public void onMapCopyCrafting(InventoryClickEvent event) {
+		if (!(event.getInventory() instanceof CraftInventoryCrafting) || event.getSlotType() != InventoryType.SlotType.RESULT)
+			return;
+
+		handleCopyProtection(event);
+	}
+
+	private void handleCopyProtection(InventoryClickEvent event) {
+		ItemStack item = event.getCurrentItem();
+
+		if (item == null || item.getType() == Material.AIR) return;
+
+		if (isCopy(item)) {
+			event.setCancelled(true);
+			event.getWhoClicked().sendMessage(
+					Component.text()
+							.color(TextColor.color(0xFF5555))
+							.content("You can not copy copy protected maps")
```

Comment:
Perhaps "You can not **duplicate** copy protected maps" to avoid the consecutive "copy"?

Reply:
You could also hyphenate: "You can not copy copy-protected maps"

Reply:
How does clone sound? It's what the recipe is actually called.